src/coverlet.core/Coverage.cs

```diff
@@ -392,7 +392,33 @@ private void CalculateCoverage()
                     }
                 }
 
-                List<(int docIndex, int line)> zeroHitsLines = new List<(int docIndex, int line)>();
+                // Calculate lines to skip for every hits start/end candidate
+                // Nested ranges win on outermost one
+                foreach (HitCandidate hitCandidate in result.HitCandidates)
+                {
+                    if (hitCandidate.isBranch || hitCandidate.end == hitCandidate.start)
+                    {
+                        continue;
+                    }
+
+                    foreach (HitCandidate hitCandidateToCompare in result.HitCandidates)
+                    {
+                        if (hitCandidate != hitCandidateToCompare && !hitCandidateToCompare.isBranch)
+                        {
+                            if (hitCandidateToCompare.start >= hitCandidate.start &&
+                               hitCandidateToCompare.end <= hitCandidate.end)
```

Comment:
Here is where I made the mistake in my branch. I just had an issue with the range :man_facepalming:. I really like your idea of saving the excluded lines in a ```HashSet<int>``` this removes the overhead of finding the top most :thumbsup:.